extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/indexing/destination/DataSourceMSQDestinationTest.java

```diff
@@ -20,17 +20,41 @@
 package org.apache.druid.msq.indexing.destination;
 
 
+import com.google.common.collect.ImmutableMap;
 import nl.jqno.equalsverifier.EqualsVerifier;
+import org.apache.druid.data.input.impl.DimensionSchema;
+import org.apache.druid.data.input.impl.StringDimensionSchema;
 import org.junit.Test;
 
+import java.util.Map;
+
 public class DataSourceMSQDestinationTest
 {
 
   @Test
   public void testEquals()
   {
     EqualsVerifier.forClass(DataSourceMSQDestination.class)
-                  .withNonnullFields("dataSource", "segmentGranularity", "segmentSortOrder")
+                  .withNonnullFields("dataSource", "segmentGranularity", "segmentSortOrder", "dimensionToSchemaMap")
+                  .withPrefabValues(
+                      Map.class,
+                      ImmutableMap.of(
+                          "language",
+                          new StringDimensionSchema(
+                              "language",
+                              DimensionSchema.MultiValueHandling.SORTED_ARRAY,
+                              false
+                          )
+                      ),
+                      ImmutableMap.of(
+                          "region",
+                          new StringDimensionSchema(
+                              "region",
+                              DimensionSchema.MultiValueHandling.SORTED_ARRAY,
+                              false
+                          )
+                      )
```

Comment:
Based on my limited understanding, it's only required if a class is self-referential. `DataSourceMSQDestination` doesn't has a reference to `DataSourceMSQDestination`, so I don't think we'd need this.

Reply:
This is required for `Map` which throws `Role's equals method delegates to an abstract method` error otherwise.